lightning/src/onion_message/messenger.rs

```diff
@@ -922,22 +989,36 @@ where
 
 				match message {
 					ParsedOnionMessageContents::Offers(msg) => {
-						let response = self.offers_handler.handle_message(msg);
-						self.handle_onion_message_response(
-							response, reply_path, format_args!(
-								"when responding to Offers onion message with path_id {:02x?}",
-								path_id
-							)
-						);
+						// message_type is defined outside of closure to avoid burrowing
+						// issues when using msg for creating ReceivedOnionMessage::New()
+						let message_type = msg.msg_type();
+						let closure = |response, reply_path: BlindedPath| {
+							self.handle_onion_message_response(
+								response, reply_path, format_args!(
+									"when responding to {} onion message with path_id {:02x?}",
+									message_type,
+									path_id
+								)
+							);
+						};
```

Comment:
I *think* we can DRY this up by moving the `closure` definition outside the `match`.

Reply:
I gave this a shot, but since the type of `response` depends on the type of `message`, creating and using the same function for both handle_message and handle_custom_message, became tricky.
So I have hit a roadblock on this idea for now!

Let me know if there's some simpler way to achieve DRY code here, and I would love to make the change!
Thanks!

Reply:
Good point! We'll have to define the closure using a local macro definition instead.